lib/pedicel/validator.rb

```diff
@@ -12,25 +18,116 @@ class TokenDataFormatError < Error; end
       hex?: 'invalid hex',
       pan?: 'invalid pan',
       yymmdd?: 'invalid date format YYMMDD',
-    }
-
+      ecPublicKey?: 'is not a EC public key',
+      PKCS7Signature?: 'is not a PKCS7 Signature',
+      ECI?: 'not an ECI indicator',
+      hexsha256?: 'not hex-encoded SHA256',
+      base64sha256?: 'not base64-encoded SHA256'
+    }.freeze
+
+    # Custom predicates for validation.
+    # Some predicates statements have been unfolded for unit-testing purposes,
+    # since _simplecov_ cannot check which branches have been used.
     module Predicates
       include Dry::Logic::Predicates
 
-      predicate(:base64?) { |value| !Base64.decode64(value).nil? rescue false }
+      predicate(:base64?) do |value|
+        begin
+          !Base64.strict_decode64(value).nil?
+        rescue StandardError => _
+          false
+        end
+      end
 
-      predicate(:hex?) { |value| !Regexp.new(/\A[a-f0-9-]*\z/i).match(value).nil? }
+      predicate(:hex?) do |value|
+        if Regexp.new(/\A[a-f0-9-]*\z/i).match(value).nil?
+          false
+        else
+          true
+        end
+      end
 
-      predicate(:pan?) { |value| !Regexp.new(/\A[0-9]{13,19}\z/).match(value).nil? }
+      # Reference: https://en.wikipedia.org/wiki/Payment_card_number
+      predicate(:pan?) do |value|
+        if Regexp.new(/\A[0-9]{12,19}\z/).match(value).nil?
+          false
+        else
+          true
+        end
+      end
 
       predicate(:yymmdd?) do |value|
-        return false unless value.length == 6
-        Time.new(value[0..1],value[2..3], value[4..5]).is_a?(Time) rescue false
+        unless value.length == 6 # rubocop:disable Style/IfUnlessModifier
```

Comment:
Do we want these rubocop magic knobs? We have not used Rubocop, at least in PCI, the last two years. Are we starting to?

Reply:
We could use codeclimate ...

Reply:
I like having some style-checker, but it's not up to me.

Is there a specific reason to use CodeClimate instead on rubocop?
Rubocop seems to work well for me. It's the ruby language that generates the errors.